mlflow/entities/metric.py

```diff
@@ -1,15 +1,33 @@
 from mlflow.entities._mlflow_object import _MLflowObject
 from mlflow.protos.service_pb2 import Metric as ProtoMetric
 
+from .metic_value_conversion_utils import (
+    convert_metric_value_to_float_if_possible,
+    convert_metric_value_to_str_if_possible,
+)
+
 
 class Metric(_MLflowObject):
     """
     Metric object.
+
```

Comment:
@ntakouris Rather than making changes to the low-level metric entity, I think we should update the fluent `mlflow.log_metric` and `mlflow.log_metrics()` APIs to support these additional types (https://github.com/mlflow/mlflow/blob/37c678320fbf82b93d2614dd14c0bee784d95175/mlflow/tracking/fluent.py#L559-L587). The fundamental definition of a metric should have a string key and a scalar numeric value, but upstream user-facing APIs can definitely accept multiple types and coerce them to this format. Please let me know if you have questions here.

Reply:
Agreed. Will migrate changes to the high level api.